### PR TITLE
erasure: listOnlineDisks should return errFileNotFound for errReadQuorum

### LINKED ISSUE #1571: Unclear error message when previously ran minio server with fresh disks
1. Ran minio server `minio server /d0 /d1 /d2 /d3`
2. Stop and rerun minio server fresh `/d1 /d2 /d3` like `minio server /d0 /d1 /d2 /d3` causes below error

```
FATA[0000] Initializing object layer failed.             Error={Cause:I/O error.  did not meet read quorum. Type:*errors.errorString}
```

Which is not clear to the user what's wrong with given arguments

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #1571

### CODE REVIEW

**Comment by krishnasrinivas** [comment]
looks good.

**Comment by krishnasrinivas** [comment]
need to review the entire erasure layer.

File: xl-erasure-v1-common.go [hunk 1]
```diff
@@ -62,12 +62,13 @@ func (xl XL) listOnlineDisks(volume, path string) (onlineDisks []StorageAPI, mda
 	for _, err := range errs {
 		if err == errFileNotFound {
 			notFoundCount++
+			// If we have errors with file not found equal to the number of disks.
+			if notFoundCount > len(xl.storageDisks)-xl.readQuorum {
+				return nil, xlMetaV1{}, false, errFileNotFound
+			}
```
Comment (abperiasamy) [inline xl-erasure-v1-common.go:68]: We should return ErrInternalError or make a new custom error code ErrIOError

**Comment by harshavardhana** [comment]
> We should return ErrInternalError or make a new custom error code ErrIOError

The reason was that `notFoundCount > len(xl.storageDisks)-xl,readQuorum` means we have exhausted all possibilities of recovery that is possibly we are better off returning 'errFileNotFound' since the actual error was 'errFileNotFound' from other disks. 

We have errReadQuorum and errWriteQuorum as separate errors for conditions where indeed there are disks not in quorum.

**Pull request merged**